### ISSUE TITLE
BUG Tab counts stopped updating on watch/unwatch

### ISSUE THREAD

**Issue by blahosyl** [blahosyl/task-manager-frontend#159]
Probably after the changes in connection with #156 

Intended way of working: #124 

Task list is updated, but watch count is not.

It works on ProfileDetail

![Image](https://github.com/user-attachments/assets/a0cc5b91-28d3-474a-a1bf-03f5fca90d6d)

but not on TaskList (neither list nor kanban)

![Image](https://github.com/user-attachments/assets/260ceb1b-430c-47e1-b9b6-ed0bcbcef46e)

**Comment by blahosyl** [comment]
Contacted tutoring, but unfortunately we couldn't solve the issue: triggering the refetching of profiles brought back the bug described in #156

**Comment by blahosyl** [comment]
On further examination, I decided to remove the useEffect hook from `TaskDetail`, and only use the one in `TaskTabs`. Adding an additional trigger to this hook, tabListChanged, fixed the Tab counts in the task list & kanban views but broke it on Profile Detail.

Comparing the hooks in `TaskTabs` & `ProfileDetail` I managed to make both work by removing `setTabListChanged(false);` from the hook in `ProfileDetail`.

Currently, task counts and task lists are updated on both views, and profile detail links work correctly (#156 ) is not triggered anymore.
